platform/config/config-mapping-impl/src/main/java/org/codice/ddf/config/mapping/impl/ConfigMappingImpl.java

```diff
@@ -0,0 +1,273 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.config.mapping.impl;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.codice.ddf.config.Config;
+import org.codice.ddf.config.ConfigEvent;
+import org.codice.ddf.config.ConfigGroup;
+import org.codice.ddf.config.ConfigService;
+import org.codice.ddf.config.ConfigSingleton;
+import org.codice.ddf.config.mapping.ConfigMapping;
+import org.codice.ddf.config.mapping.ConfigMappingException;
+import org.codice.ddf.config.mapping.ConfigMappingProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigMappingImpl implements ConfigMapping {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigMappingImpl.class);
+
+  // marker to indicate that a config mapping depends on all instances/ids of a given config type
+  private static final Set<String> ALL = Collections.emptySet();
+
+  // marker to representing a fake instance id representing any instances
+  private static final String ANY = "*";
+
+  private final ConfigService config;
+
+  private final ConfigMapping.Id id;
+
+  private final SortedSet<ConfigMappingProvider> providers;
+
+  // keyed by config type with corresponding set of instances or ALL if not a group or dependent on
+  // all instances
+  private final Map<Class<? extends Config>, Set<String>> dependents = new ConcurrentHashMap<>();
+
+  public ConfigMappingImpl(
+      ConfigService config, ConfigMapping.Id id, Stream<ConfigMappingProvider> providers) {
+    this.config = new DependentConfigService(config);
+    this.id = id;
+    this.providers =
+        Collections.synchronizedSortedSet(providers.collect(Collectors.toCollection(TreeSet::new)));
+    try {
+      // first resolution to compute the initial dependents, use the ANY instance if none defined
+      // just in case the providers are referencing it. This will be useful providers capable of
+      // providing for any instances
+      resolve(ConfigMapping.Id.of(id.getName(), id.getInstance().orElse(ConfigMappingImpl.ANY)));
+    } catch (ConfigMappingException e) { // ignore
```

Comment:
We could. I just didn't since this is just to pre-initialize things. The real problem will be later when the public `resolve()` is actually called and I will leave it up to the clients to determine if it should be logged or not. 

I also feel that the providers themselves might be logging so I didn't want to double log for no additional reasons here.